pkg/client/client.go

```diff
@@ -0,0 +1,179 @@
+// Package client provides a client interface to the Agent HTTP
+// API.
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/agent/pkg/prometheus/ha/configapi"
+	"github.com/grafana/agent/pkg/prometheus/instance"
+	"gopkg.in/yaml.v2"
+)
+
+// Client is a collection of all subsystem clients.
+type Client struct {
+	PrometheusClient
+}
+
+// New creates a new Client.
+func New(addr string) *Client {
+	return &Client{
+		PrometheusClient: &prometheusClient{addr: addr},
+	}
+}
+
+// PrometheusClient is the client interface to the API exposed by the
+// Prometheus subsystem of the Grafana Cloud Agent.
+type PrometheusClient interface {
+	// Instances runs the list of currently running instances.
+	Instances(ctx context.Context) ([]string, error)
+
+	// The following methods are for the scraping service mode
```

Comment:
Maybe an improvement for the future (create an issue?) is that we let the user know. We could catch 404s and reword the error to something like "looks like you have a wrong address for the agent or do not have scraping service mode enabled".

Reply:
An even better alternative is to have the route always mounted and return the descriptive error from itself.

Reply:
:+1: I created #64 for this.